pkg/kubelet/util/env/env_util_test.go

```diff
@@ -36,245 +68,483 @@ func TestParseEnv(t *testing.T) {
+			name: "unquoted value triggers error",
+			envContent: `KEY1='foo'
+KEY2=bar
+KEY3='baz'
 `,
-			key:         "KEY1",
+			key:         "KEY2",
 			wantErr:     true,
-			errContains: "unfinished line continuation",
+			errContains: "value must be enclosed in single quotes",
 		},
 		{
-			name: "key not found returns empty",
-			envContent: `KEY1=foo
-KEY2=bar
+			name: "Double quoted value triggers error",
+			envContent: `KEY1='foo'
+KEY2="bar"
+KEY3='baz'
 `,
-			key:       "KEY3",
-			wantValue: "",
+			key:         "KEY2",
+			wantErr:     true,
+			errContains: "value must be enclosed in single quotes",
 		},
 		{
-			name: "value with embedded #",
-			envContent: `KEY1=foo#notcomment
-KEY2=bar
+			name: "unclosed single quote triggers error",
+			envContent: `KEY1='foo'
+KEY2='bar
+KEY3='baz'
+`,
+			key:         "KEY2",
+			wantErr:     true,
+			errContains: "unexpected content after closing quote",
+		},
+		{
+			name: "content after closing quote triggers error",
+			envContent: `KEY1='foo'
+KEY2='bar'baz
+KEY3='baz'
+`,
+			key:         "KEY2",
+			wantErr:     true,
+			errContains: "unexpected content after closing quote",
+		},
+		{
+			name: "empty quotes preserve literal content",
+			envContent: `KEY1=''
+KEY2='   '
+KEY3='bar'
 `,
 			key:       "KEY1",
-			wantValue: "foo#notcomment",
+			wantValue: ``,
 		},
 		{
-			name: "key with trailing whitespace",
-			envContent: `KEY1 =foo
-KEY2=bar
+			name: "quotes preserve all literal content including spaces",
+			envContent: `KEY1='  foo  bar  '
+KEY2='bar'
 `,
 			key:       "KEY1",
-			wantValue: "foo",
+			wantValue: `  foo  bar  `,
 		},
 		{
-			name: "value with leading and trailing whitespace",
-			envContent: `KEY1=   foo bar   
-KEY2=bar
+			name: "special characters in value",
+			envContent: `KEY1='!@#$%^&*()_+-=[]{}|;:,.<>?/'
+KEY2='bar'
 `,
 			key:       "KEY1",
-			wantValue: "foo bar",
+			wantValue: `!@#$%^&*()_+-=[]{}|;:,.<>?/`,
 		},
 		{
-			name: "multiple comments and blank lines",
-			envContent: `# first comment
-
-# second comment
-KEY1=foo
-
-# third comment
-KEY2=bar
+			name: "newlines and tabs in value",
+			envContent: `KEY1='line1\nline2\tline3'
+KEY2='bar'
 `,
-			key:       "KEY2",
-			wantValue: "bar",
+			key:       "KEY1",
+			wantValue: `line1\nline2\tline3`,
 		},
 		{
-			name: "continuation with blank line in between (should error)",
-			envContent: `KEY1=foo \
-
-bar
+			name: "unicode characters in value",
+			envContent: `KEY1='中文 Español Français 🌍'
+KEY2='bar'
 `,
-			key:         "KEY1",
-			wantErr:     true,
-			errContains: "invalid environment variable format",
+			key:       "KEY1",
+			wantValue: `中文 Español Français 🌍`,
 		},
 		{
-			name: "continuation with comment in between (should error)",
-			envContent: `KEY1=foo \
-# comment
-bar
+			name: "backslashes in value",
+			envContent: `KEY1='path\\to\\file'
+KEY2='bar'
+`,
+			key:       "KEY1",
+			wantValue: `path\\to\\file`,
+		},
+		{
+			name: "quotes within quotes",
```

Comment:
nit: Since double quotes are allowed, it might be fine to explicitly use "single quotes" instead.

```suggestion
			name: "single quotes within value quotes",
```

In the `multi-line value with code-like content` test, there’s a test case that uses double quotes, so it’s not strictly required, but it might be better to add a test case for double quotes after the one for single quotes.

```go
		{
			name: "double quotes within value quotes",
			envContent: `KEY1='value with \"nested\" quotes'
KEY2='bar'
`,
			key:       "KEY1",
			wantErr:   false,
			wantValue: `value with \"nested\" quotes`,
		},
```